### PR TITLE
prov/gni: Add CCM capability

### LINKED ISSUE #663: Get GNI provider working under CCM

### PR DESCRIPTION
**Pull request by jshimek**
Provide ability to read the ccm file, so when running
using ccm mode you can use the gni provider.

Signed-off-by: James Shimek jshimek@cray.com
@sungeunchoi @hppritcha @jswaro 

fixes #663

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Test FAILed.

File: prov/gni/src/gnix_util.c [hunk 1]
```diff
@@ -243,6 +243,36 @@ static int gnix_alps_init(void)
...
+int get_ccm_ptag_cookie(char *filename)
+{
+	int rc, fd;
+	uint32_t version;
+	ccm_alps_info_t info;
+	GNIX_INFO(FI_LOG_FABRIC, "Reading job info file %s", filename);
+
+	fd = open(filename, O_RDONLY);
+	if (fd < 0)
+		return -errno;
```
Comment (ztiffany) [inline prov/gni/src/gnix_util.c]: Use FI_E\* return codes if possible

File: prov/gni/src/gnix_util.c [hunk 2]
```diff
@@ -243,6 +243,36 @@ static int gnix_alps_init(void)
...
+	rc = read(fd, &info, sizeof(ccm_alps_info_t));
+	if (rc != sizeof(ccm_alps_info_t))
+		return -EIO;
+
+	gnix_app_ptag = info.ptag;
+	gnix_app_cookie = info.cookie;
+	version = info.version;
+
+	close(fd);
+	GNIX_INFO(FI_LOG_FABRIC, "Ptag=0x%x, cookie=0x%x", gnix_app_ptag, gnix_app_cookie);
```
Comment (ztiffany) [inline prov/gni/src/gnix_util.c]: Wrap lines over 80 characters.

File: prov/gni/src/gnix_util.c [hunk 3]
```diff
@@ -250,13 +280,15 @@ int gnixu_get_rdma_credentials(void *addr, uint8_t *ptag, uint32_t *cookie)
...
 	if (ret) {
-		GNIX_WARN(FI_LOG_FABRIC,
-			  "gnix_alps_init() failed, ret=%d(%s)\n",
-			  ret, strerror(errno));
-		return ret;
+		ret = gnix_alps_init();
+		if (ret) {
+			GNIX_WARN(FI_LOG_FABRIC,
+				  "gnix_alps_init() failed, ret=%d(%s)\n",
+			  	ret, strerror(errno));
```
Comment (ztiffany) [inline prov/gni/src/gnix_util.c]: Misaligned indent

File: prov/gni/src/gnix_util.c (hunk 1 above)
Comment (sungeunchoi) [inline prov/gni/src/gnix_util.c]: I ran into this issue with the kdreg stuff.  Any thoughts on what to do generally speaking?  Not all the possible errnos are defined ad FI_E\* codes.  I put in FI_WARN's with strerror(), to help with that, but it's still not great.

Comment (ztiffany) [inline prov/gni/src/gnix_util.c]: Generally, I don't think we need to translate every error into a unique FI_\* errno value.  In this case I think any open/read/write errors could result in get_ccm_ptag_cookie() returning -FI_EIO.

**Comment by hppritcha** [comment]
looks good to me :+1:

**Pull request merged**